src/options-component/DataTab.js

```diff
@@ -0,0 +1,174 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from 'material-ui-next/styles';
+import classNames from 'classnames';
+import { FormControl } from 'material-ui-next/Form';
+import Select from 'material-ui-next/Select';
+import TextField from 'material-ui-next/TextField';
+import { InputLabel } from 'material-ui-next/Input';
+import { MenuItem } from 'material-ui-next/Menu';
+import DataTabCheckBoxes from './DataTabCheckBoxes';
+import strings from './utils';
+
+const styles = {
+    formControl: {
+        minWidth: 200,
+    },
+    inputLabeltextSize: {
+        fontSize: 13,
+    },
+    textFields: {
+        marginRight: 20,
+        width: 100,
+    },
+    formControlCoverWholeRow: {
+        marginRight: '100%',
+    },
+    textFieldCoverWholeRow: {
+        marginRight: '55%',
+        width: 200,
+    },
+    divBorder: {
+        borderBottom: '1px solid rgb(158, 158, 158)',
+        paddingBottom: 35,
+    },
+};
+
+const renderTextFields = (classes, tabContent, onChange) => {
+    return tabContent.map(([key, value], i) => (
+        <TextField
+            className={
+                i % 2 === 0
+                    ? classes.textFields
+                    : classes.textFieldCoverWholeRow
+            } // 2nd and 4th TextField component pads out rest of the row space
+            InputLabelProps={{ className: classes.inputLabeltextSize }}
+            label={strings.data[key]}
+            onChange={event => {
+                onChange(key, event.target.value);
+            }} // TODO (?) use parseInt on number? onChange will always pass a string
+            type={i % 2 === 0 ? 'number' : 'string'} // 1st and 3rd TextField component are number specific
+            value={value}
+            key={i}
+        />
+    ));
+};
+
+const renderSelectFields = (classes, tabContent, onChange) => {
+    //render first two select components
+    // call			{renderTextFields(classes, textFields, onChange)} (?)
+    //render last select component
+};
+
+const DataTab = props => {
+    const { classes, tabContent, onChange } = props;

```

Comment:
You can write this in a more shorthand way on line 63 (and drop line 64):

const DataTab = ({ classes, tabContent, onChange }) = {

It's pretty clean as long as there aren't too many props. More than 4, then it might be better to do it as you had done.